mewarpx/tests/test_custom_criteria.py

```diff
@@ -0,0 +1,108 @@
+"""Test the custom criteria using capacitive discharge"""
+import pytest
+import numpy as np
+
+from mewarpx import util as mwxutil
+
+def check_particle_nums():
+    from mewarpx import mwxrun
+    return mwxrun.mwxrun.get_npart() < 2500000
+
+def add_particles():
+    from pywarpx import _libwarpx
+    # add 500,000 particles
+    _libwarpx.add_particles(x=np.zeros((500000)), y=np.zeros((500000)), z=np.zeros((500000)))
+
+@pytest.mark.parametrize(
+    ("name"),
+    [
+        'Run2D',
+        'Run2D_RZ',
+        'Run3D'
+    ]
+)
+def test_capacitive_discharge_multigrid(capsys, name):
+    basename = "Run"
+    use_rz = 'RZ' in name
+    dim = int(name.replace(basename, '')[0])
+
+    # Initialize and import only when we know dimension
+    mwxutil.init_libwarpx(ndim=dim, rz=use_rz)
+    from mewarpx import testing_util
+    from mewarpx.setups_store import diode_setup
+    from mewarpx import mwxrun
+    from pywarpx import callbacks
+
+    # Include a random run number to allow parallel runs to not collide. Using
+    # python randint prevents collisions due to numpy rseed below
+    #testing_util.initialize_testingdir(name)
+
+    # Initialize each run with consistent, randomly-chosen, rseed. Use a random
+    # seed instead for initial dataframe generation.
+    # np.random.seed()
+    np.random.seed(92160881)
+
+    sim = mwxrun.mwxrun.simulation
+
+    # Specific numbers match older run for consistency
+    FREQ = 13.56e6  # MHz
+    DT = 1.0 / (400 * FREQ)
+    DIAG_STEPS = 2
+    DIAG_INTERVAL = DIAG_STEPS*DT
+    VOLTAGE = 450.0
+    D_CA = 0.067  # m
+    run = diode_setup.DiodeRun_V1(
+        dim=dim,
+        rz=use_rz,
+        V_ANODE_CATHODE=VOLTAGE,
+        V_ANODE_EXPRESSION="%.1f*sin(2*pi*%.5e*t)" % (VOLTAGE, FREQ),
+        D_CA=D_CA,
+        INERT_GAS_TYPE='He',
+        N_INERT=9.64e20,  # m^-3
+        T_INERT=300.0,  # K
+        PLASMA_DENSITY=2.56e14,  # m^-3
+        T_ELEC=30000.0,  # K
+        NX=16,
+        NZ=128,
+        # This gives equal spacing in x & z
+        PERIOD=D_CA * 16 / 128.0,
```

Comment:
See NX and NY predefined at https://github.com/ModernElectron/WarpX/blob/b2a85ee6fb43e4a58a4f616388f5e74f33c118ae/mewarpx/tests/test_capacitive_discharge.py#L45 please use that.

In general please do a diff against the `test_capacitive_discharge` in `memaster` to make sure you pick up all changes from there.